sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeSuite.scala

```diff
@@ -145,8 +145,8 @@ class DataTypeSuite extends SparkFunSuite {
     val message = intercept[SparkException] {
       left.merge(right)
     }.getMessage
-    assert(message.equals("Failed to merge fields 'b' and 'b'. " +
-      "Failed to merge incompatible data types FloatType and LongType"))
+    assert(message === "Failed to merge fields 'b' and 'b'. " +
```

Comment:
Is this change a valid improvement? I forgot about `===` when I pushed that commit. I can revert this back but do I need create another PR? With or without JIRA?